projects/storefrontlib/src/cms-components/user/login-form/login-form.component.ts

```diff
@@ -8,40 +8,60 @@ import {
 } from '@spartacus/core';
 import { Subscription } from 'rxjs';
 import { CustomFormValidators } from '../../../shared/utils/validators/custom-form-validators';
+import { ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'cx-login-form',
   templateUrl: './login-form.component.html',
 })
 export class LoginFormComponent implements OnInit, OnDestroy {
-  sub: Subscription;
+  private tokenExists = false;
+  subs: Subscription;
   form: FormGroup;
 
   constructor(
     private auth: AuthService,
     private globalMessageService: GlobalMessageService,
     private fb: FormBuilder,
-    private authRedirectService: AuthRedirectService
+    private authRedirectService: AuthRedirectService,
+    private route: ActivatedRoute
```

Comment:
That is a breaking change. Let's make this service optional and if this service will be provided then support email autofill.